src/Streamly/Internal/Data/Stream/StreamD.hs

```diff
@@ -3392,22 +3392,22 @@ _bracket bef aft bet = Stream step' BracketInit
 
 -- | Run a side effect whenever the stream stops normally or aborts due to an
 -- exception.
-{-# INLINE finally #-}
-finally :: MonadCatch m => m b -> Stream m a -> Stream m a
+{-# INLINE finally_ #-}
+finally_ :: MonadCatch m => m b -> Stream m a -> Stream m a
 -- finally action xs = after action $ onException action xs
-finally action xs = bracket (return ()) (\_ -> action) (const xs)
+finally_ action xs = bracket_ (return ()) (\_ -> action) (const xs)
 
-{-# INLINE finallyIO #-}
-finallyIO :: (MonadAsync m, MonadCatch m) => m b -> Stream m a -> Stream m a
-finallyIO action xs = bracketIO (return ()) (\_ -> action) (const xs)
+{-# INLINE finally #-}
+finally :: (MonadAsync m, MonadCatch m) => m b -> Stream m a -> Stream m a
+finally action xs = bracket (return ()) (\_ -> action) (const xs)
```

Comment:
Do we need to use `MonadAsync`? `MonadCatch` implies `MonadThrow` but its included in `MonadAsync` again, probably fine. This happens in quite a few places btw.